test-data/unit/check-narrowing.test

```diff
@@ -1910,3 +1910,16 @@ if len(x) == a:
 else:
     reveal_type(x)  # N: Revealed type is "Union[Tuple[builtins.int, builtins.int], Tuple[builtins.int, builtins.int, builtins.int]]"
 [builtins fixtures/len.pyi]
+
+[case testNarrowingLenUnionWithUnreachable]
+from typing import Union, Sequence
+
+def f(x: Union[int, Sequence[int]]) -> None:
+    if (
+        isinstance(x, tuple)
+        and len(x) == 2
+        and isinstance(x[0], int)
+        and isinstance(x[1], int)
+    ):
+        reveal_type(x)  # N: Revealed type is "Tuple[builtins.int, builtins.int]"
```

Comment:
Maybe add `else: reveal_type(x)`? It might give some unexpected result.

Reply:
The else branch reveals `Union[builtins.int, typing.Sequence[builtins.int], builtins.tuple[Any, ...]]` because of the behavior I mentioned in the description. I intentionally didn't add this, since I don't want to create an impression this is intentional.